test/mocks/server/fatal_action_factory.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "envoy/server/fatal_action_config.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+class MockFatalActionFactory : public FatalActionFactory {
+public:
+  MockFatalActionFactory();
```

Comment:
You can declare these `= default` here or just omit them and let the compiler define them automatically.